### ISSUE TITLE
non-script files should not have shebang lines or be executable

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2211]
|<img alt="thomasvs's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| thomasvs reported|
|-|-|
|Trac ID|trac#2211|
|Type|enhancement|
|Created|2006-11-01 15:02:27Z|

from rpmlint:
```
E: python-twisted-web non-executable-script /usr/lib64/python2.4/site-packages/twisted/web/test/test_cgi.py 0644
E: python-twisted-web non-executable-script /usr/lib64/python2.4/site-packages/twisted/web/test/test_distrib.py 0644

```

Attachments:

* [shebang_2211.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e64/e6402ee50e78b6141db94c840ca7903762665732/46a84188a4f407915bcf590d67ec486b404eb01d.diff) (893 bytes) - added by therve on 2007-01-19 22:16:18Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2211 2211
type__enhancement enhancement
reporter__thomasvs thomasvs
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1162393347000000 1162393347000000
changetime__1169918149000000 1169918149000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="thomasvs's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|thomasvs set owner to @jyknight|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve removed owner|
|-|-|

Let's remove this.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @therve|
|-|-|

It looks like twisted/internet/glib2reactor.py also has a #! line and shouldn't.  I suppose rpmlint didn't complain about it because it is set svn:executable, along with win32eventreactor.py.  Both these properties should be removed.  Put these changes and the patch into a branch, I guess, and I'll give it another quick review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Thansk for your comment, ready to review in no-shebang-2211.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @therve|
|-|-|

Great.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [19488]) Merge no-shebang-2211.

Author: therve
Reviewer: exarkun
Fixes [#2211](https://github.com/twisted/twisted/issues/2211)

Remove shebang lines from non-scripts files, and svn:executable properties for
non executables scripts.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
